Source/data/common.hpp

```diff
@@ -0,0 +1,226 @@
+ * HandleRecordSeparator for a description of the different codes.
+ * @param begin start of a stream (expected to be pointing to a record separator)
+ * @param end one past the last character of the stream
+ * @return a struct containing a pointer to the start of the next field (if more characters are
+ *          available) or a copy of end, and optionally an error code describing what type of
+ *          separator was found.
+ */
+inline GetFieldResult HandleFieldSeparator(const char *begin, const char *end)
+{
+	if (begin != end && *begin == '\t') {
+		return { begin + 1 };
+	}
+
+	return HandleRecordSeparator(begin, end);
+}
+
+/**
+ * @brief Advances to the next field separator without saving any characters
+ * @param begin first character of the stream
+ * @param end one past the last character in the stream
+ * @return a GetFieldResult struct containing the start of the next field/record and an error code if the last field in the record was found
+ */
+inline GetFieldResult DiscardField(const char *begin, const char *end)
+{
+	const char *nextSeparator = std::find_if(begin, end, IsFieldSeparator);
+
+	return HandleFieldSeparator(nextSeparator, end);
+}
+
+/**
+ * @brief Advances by the specified number of fields or until the end of the record, whichever occurs first
+ * @param begin first character of the stream
+ * @param end one past the last character in the stream
+ * @param skipLength how many records to skip (specifying 0 will cause the method to return without advancing)
+ * @return a GetFieldResult struct containing the start of the next field/record and an error code if the last field in the record was found
+ */
+GetFieldResult DiscardMultipleFields(const char *begin, const char *end, unsigned skipLength);
+
+/**
+ * @brief Reads a sequence of character bytes into the provided buffer from a tab-delimited stream.
+ *
+ * Note that buffer is *always* modified by this function as a zero-length field is a valid value.
+ * This function consumes the field separator whenever possible, the `ptr` member of the returned
+ * type will be either the start of the next field/record or `end`. The `ec` member contains
+ * additional information to distinguish between the end of a field and the end of a record. If
+ * there are additional fields in this record then `ec` will be value initialised
+ * (`ec` == DataFieldError()), otherwise refer to HandleRecordSeparator for the meanings associated
+ * with different codes.
+ * @param begin first character of the stream
+ * @param end one past the last character in the stream
+ * @param buffer output for the field contents, will be filled with a copy of the characters between [begin, fieldEnd)
+ * @return a GetFieldResult struct containing the start of the next field/record and an error code if the last field in the record was found
+ */
+inline GetFieldResult GetNextField(const char *begin, const char *end, std::string &buffer)

```

Comment:
Perhaps this should return an `std::string_view` in `GetFieldResult`? There is no need to copy the string here from what I can tell.

Reply:
I think I did this to mirror the API of `from_chars`. There's no real point doing that for a string to string copy so I'll add a member to the result type instead of using an output param.